### PR TITLE
Add key history iterator

### PR DESCRIPTION
**Pull request by e-e-e**
This is a simple PR to add the ability to stream the history of a specific prefix.

I want to be able to quickly iterate all values that a key has ever been.

I have write basic tests and it appears to be working, but I wanted to get your thoughts on this @mafintosh, to see if I am approaching the problem in the best way - or if there is a smarter way of traversing backwards.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -506,8 +507,12 @@ HyperDB.prototype.list = function (prefix, opts, cb) {
   }
 }
 
-HyperDB.prototype.history = function (opts) {
-  return history(this, opts)
+HyperDB.prototype.history = function (prefix, opts) {
+  if (prefix && !isOptions(prefix)) {
+    console.log('KEY')
```
Comment (mafintosh) [inline index.js]: prob didn't mean to add this?

**Comment by mafintosh** [comment]
This looks good to me, but I wonder if this would be better as a new api instead of being in the history one, as it's kinda different behaivor right?

File: index.js (hunk 1 above)
Comment (e-e-e) [reply index.js]: Ah no I totally did not - good catch. Thanks!

**Comment by e-e-e** [comment]
Hey @mafintosh - I dont think @pfrazee had any particular naming suggestions.
Is it possible to get this merged? I would love to implement versioning in hyper-readings but require this functionality to do so.

**Comment by mafintosh** [comment]
Great job! Especially with the tests

**Comment by mafintosh** [comment]
3.3.0 :tada:

**Comment by pfrazee** [comment]
Thanks @e-e-e !

**Pull request merged**